ksql-rest-app/src/test/java/io/confluent/ksql/rest/server/execution/InsertValuesExecutorTest.java

```diff
@@ -0,0 +1,387 @@
+import io.confluent.ksql.statement.ConfiguredStatement;
+import io.confluent.ksql.util.KsqlConfig;
+import io.confluent.ksql.util.KsqlException;
+import io.confluent.ksql.util.timestamp.MetadataTimestampExtractionPolicy;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import org.apache.kafka.clients.producer.KafkaProducer;
+import org.apache.kafka.clients.producer.ProducerRecord;
+import org.apache.kafka.common.serialization.Serde;
+import org.apache.kafka.common.serialization.Serializer;
+import org.apache.kafka.connect.data.Field;
+import org.apache.kafka.connect.data.Schema;
+import org.apache.kafka.connect.data.SchemaBuilder;
+import org.apache.kafka.streams.KafkaClientSupplier;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+
+@RunWith(MockitoJUnitRunner.class)
+public class InsertValuesExecutorTest {
+
+  private static final Schema SCHEMA = SchemaBuilder.struct()
+      .field("ROWTIME", Schema.OPTIONAL_INT64_SCHEMA)
+      .field("ROWKEY", Schema.OPTIONAL_INT64_SCHEMA)
+      .field("COL0", Schema.OPTIONAL_INT64_SCHEMA)
+      .field("COL1", Schema.OPTIONAL_STRING_SCHEMA)
+      .build();
+
+  private static final Schema STRICT_SCHEMA = SchemaBuilder.struct()
+      .field("ROWTIME", Schema.INT64_SCHEMA)
+      .field("ROWKEY", Schema.INT64_SCHEMA)
+      .field("COL0", Schema.OPTIONAL_INT64_SCHEMA)
+      .field("COL1", Schema.OPTIONAL_STRING_SCHEMA)
+      .build();
+
+  private static final byte[] KEY = new byte[]{1};
+  private static final byte[] VALUE = new byte[]{2};
+
+  private static final String TOPIC_NAME = "topic";
+
+  @Rule
+  public final ExpectedException expectedException = ExpectedException.none();
+
+  @Mock
+  private KsqlEngine engine;
+  @Mock
+  private KsqlTopicSerDe topicSerDe;
+  @Mock
+  private Serde<String> keySerDe;
+  @Mock
+  private Serializer<String> keySerializer;
+  @Mock
+  private Serde<GenericRow> rowSerDe;
+  @Mock
+  private Serializer<GenericRow> rowSerializer;
+  @Mock
+  private ServiceContext serviceContext;
+  @Mock
+  private KafkaProducer<byte[], byte[]> producer;
+
+  @Before
+  public void setup() {
+    when(topicSerDe.getGenericRowSerde(any(), any(), any(), any(), any())).thenReturn(rowSerDe);
+
+    when(keySerDe.serializer()).thenReturn(keySerializer);
+    when(rowSerDe.serializer()).thenReturn(rowSerializer);
+
+    when(keySerializer.serialize(any(), any())).thenReturn(KEY);
+    when(rowSerializer.serialize(any(), any())).thenReturn(VALUE);
+
+    final KafkaClientSupplier kafkaClientSupplier = mock(KafkaClientSupplier.class);
+    when(kafkaClientSupplier.getProducer(any())).thenReturn(producer);
+
+    when(serviceContext.getKafkaClientSupplier()).thenReturn(kafkaClientSupplier);
+
+    givenDataSourceWithSchema(SCHEMA);
+  }
+
```

Comment:
We should have an explicit key for handling omitted columns by setting them to null